seps/1932-dpop-extension.md

```diff
@@ -0,0 +1,165 @@
+
+- **Status**: Draft
+- **Type**: Standards Track
+- **Created**: 2025-12-05
+- **Author(s)**: Pieter Kasselman
+- **Sponsor**: Darin McAdams
+- **PR**: https://github.com/modelcontextprotocol/modelcontextprotocol/pull/1932
+
+## Abstract
+
+This SEP defines an optional DPoP (Demonstrating Proof of Possession) extension for the Model Context Protocol to support sender-constrained access tokens. The extension binds OAuth 2.0 access tokens to cryptographic key pairs controlled by MCP clients, requiring clients to demonstrate possession of the corresponding private key with each request. To align DPoP with MCP’s single-endpoint architecture, the extension incorporates a content digest into the DPoP proof, allowing the proof to be tied to the specific JSON-RPC request body and ensuring tighter request-level binding.
+
+## Motivation
+
+MCP’s current authorization model uses bearer tokens, which can be reused by an unauthorized party if the token is leaked or intercepted. If an access token is intercepted—through network eavesdropping, compromised logs, or other means—an attacker can use it to access protected MCP resources until the token expires.
+
+DPoP addresses this by making tokens "sender-constrained": even if an attacker intercepts an access token, they cannot use it without also possessing the client's private key. This significantly raises the bar for attackers.
+
+MCP's architecture presents a unique security challenge: all JSON-RPC requests use the same HTTP endpoint and method (POST). Standard DPoP binds proofs to the endpoint and HTTP method but not to the request payload, meaning intercepted proofs can be replayed with a different message body within the validity window. This SEP addresses this by including a content digest in DPoP proofs, binding each proof cryptographically to a specific request body.
+
+This extension is particularly valuable for:
+
+- High-security environments handling sensitive data
+- Long-lived access tokens
+- Deployments where token theft risk is elevated
+- Compliance requirements mandating proof of possession
+
+## Specification
+
+The DPoP Profile for MCP adapts OAuth 2.0 sender-constrained tokens for use in the Model Context Protocol’s single-endpoint architecture. Standard DPoP binds proofs to the HTTP method and URL, but in MCP all requests use the same method (POST) and endpoint, with the specific operation conveyed in the JSON-RPC body. As a result, traditional DPoP does not distinguish between different MCP operations, since the request semantics are not reflected in the elements covered by the proof. The profile refines the content of the DPoP proof to better account for this characteristic of MCP.
+
+To solve this, the profile adds one key requirement: DPoP proofs must include a cryptographic digest of the JSON-RPC request body (`content_digest`). This binds each proof to a specific request payload, preventing attackers from replaying valid proofs with altered bodies.
```

Comment:
I keep referring [to the RFC](https://datatracker.ietf.org/doc/html/rfc9449#name-request-integrity) and I am not sure why we need to do things differently for MCP here. The RFC does leave room for extensibility, I am trying to see what we'd be gaining here in terms of attack prevention that is not covered by baseline DPoP.